tunnel/router/router_linux.go

```diff
@@ -117,17 +117,16 @@ func nlAddrReq(localPrefix, peerPrefix *net.IPNet, ifName string, t netlink.Head
 		Data: append(ifmBytes, attrBytes...),
 	}
 
-	msgs, err := c.Execute(req)
-	if err == unix.EEXIST {
-		return nil
-	}
 
-	n := len(msgs)
-	if n != 1 {
-		return fmt.Errorf("expected 1 netlink message, but got %d", n)
+	_, err = c.Execute(req)
+	if err != nil {
```

Comment:
Do you need to check if the cast succeeded, or is Execute guaranteed to return errors of this type?